sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SyntaxTrackers/CSharpFieldAccessTracker.cs

```diff
@@ -44,8 +44,8 @@ public CSharpFieldAccessTracker(IAnalyzerConfiguration analyzerConfiguration, Di
         protected override GeneratedCodeRecognizer GeneratedCodeRecognizer { get; } =
             CSharp.GeneratedCodeRecognizer.Instance;
 
-        protected override SyntaxNode GetIdentifier(SyntaxNode expression) =>
-            ((ExpressionSyntax)expression).GetIdentifier();
+        protected override string GetFieldName(SyntaxNode expression) =>
+            ((ExpressionSyntax)expression).GetIdentifier()?.Identifier.ValueText;
```

Comment:
[minor] I am playing belt and braces here but I think it would be better to do a `as` cast.

Reply:
Or can we force the type to be `ExpressionSyntax`?

Reply:
The `expression` is the value of `context.Node`, which is the syntax representing the SyntaxKind enumerations above. Generally it should not throw "randomly", because if you add more SyntaxKind items, you will "usually" write a test...